src/terminal.rs

```diff
@@ -0,0 +1,78 @@
+use std::io::{Stdin, Stdout, Write};
+use crate::terminalerror::TerminalError;
+use crate::todo::Todo;
+use console::style;
+
+pub struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    pub fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn input(&mut self) -> Result<String, TerminalError> {
+        let mut buf: String = String::new();
+
+        match self.stdin.read_line(&mut buf) {
+            Ok(_) => Ok(buf.trim().to_string()),
+            Err(error) => Err(TerminalError::Stdin(error))
+        }
+    }
+    fn should_ask_for_new_todo(&mut self) -> Result<bool, TerminalError> {
+        let mut res = self.input()?;
+        while res.to_lowercase() != "sim" && res.to_lowercase() != "nao" {
+            println!("{}", style("COMANDO ERRADO").red());
+            println!("Digite {} ou {}", style("Sim").blue().bold(), style("Nao").yellow().bold());
+            res = self.input()?;
+        }
+        if res.to_lowercase() == "sim" {
+            Ok(true)
+        } else {
+            Ok(false)
+        }
+        
+    }
+
+    fn ask_for_new_todo(&mut self) -> Result<Option<Todo>, TerminalError> {
+        println!("{}", style("Olá, deseja adicionar um novo ToDo? ").green());
+        println!("[{}|{}]", style("Sim").bold().blue(), style("Nao").bold().yellow());
+
+        let res = self.should_ask_for_new_todo()?;
+        if res {
```

Comment:
Nesses casos, vc n precisa da variável.
```suggestion
        if self.should_ask_for_new_todo()? {
```